packages/nitro-protocol/contracts/ETHAssetHolder.sol

```diff
@@ -62,6 +67,7 @@ contract ETHAssetHolder is AssetHolder {
      * @param amount Quantity of wei to be transferred.
      */
     function _transferAsset(address payable destination, uint256 amount) internal override {
-        destination.transfer(amount);
+        (bool success, ) = destination.call{value: amount}('');
```

Comment:
This is actually problematic. Because the `.transfer()` method was preventing reentrancy by only giving the callee the 2300 gas stipend. We should either keep this as was or add a reentrancy guard somehow.

Reply:
I was following this advice https://diligence.consensys.net/blog/2019/09/stop-using-soliditys-transfer-now/. 

It seems best to not rely on the gas stipend for safety. We currently use a checks-effects-interaction pattern to protect against reentrancy attacks in the `AssetHolder` base contract. If we think an explicit reentrancy guard is a good idea (I don't think it is necessary), it would probably make most sense to put it in the base contract, too.